### ISSUE TITLE
[FEATURE REQUEST] Add one (or more) gift for each purchased item

### ISSUE THREAD

**Issue by zazy** [code4business/freeproduct#30]
I have these skus in the catalog: PROD1 and GIFT1.
If I purchase PROD1 I'll get GIFT1 for free. If I purches 2 PROD1 I sould get 2 GIFT1 for free and so on. 
To accomplish this task I have a shopping cart rule so that if there is an item in the cart with sku PROD1 so add GIFT1. With this rule the quantity field in shopping cart is ignored. To let it work I should add a rule for each possible value for the field QUANTITY. So I should have rules like "if there is sku PROD1 and Qty is 1 add 1 GIFT1", "if there is sku PROD1 and Qty is 2 add 2 GIFT1" and so on. Suppose there are 10 skus of this kind, if I want this to work up to 10 quantity for each sku I should add 100 rules.
Hope it's clear. Thank you for your attention.

**Comment by domeglic** [comment]
The problem I see with this is how to implement the configuration for it. The rule can often have conditions for multiple products where this doesn't make sense because you would have lots of gift products added.

The only case where it would work from what I see is if the conditions apply it to a single product. (SKU is X).

**Comment by zazy** [comment]
Hi domeglic, thank you for your reply. After writing my comment I found a mswebdesign fork of your project. Commit 01d2586 introduces exactly the feature I'm talking about. I merged it into my copy and testing it. It appears to do the job.

The use case is exactly the one you described. I'm using it to fill a gap in Magento that makes it impossible to create something like "Christmas gift basket", a set of products sell as is with no ability to change anithing and with a fixed price that may or may not be the sum of the products price. So I created a "dummy" product (a box of products) and with freeproduct I add every product contained in the box. This way customer can see every single product and related quantities that is purchasing directly in the cart.
